src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Cache.cs

```diff
@@ -248,9 +248,12 @@ private static void Add(Key key, Regex regex)
                         }
                     }
 
-                    // Remove the key found to have the smallest access stamp.
+                    // Remove the key found to have the smallest access stamp. List ordering isn't important, so rather than
+                    // just removing the element at minListIndex, which would result in an O(N) shift down, we copy the last
```

Comment:
"shift up" instead of "shift down"?

In either case, this makes sense to avoid the perf hit of the copy\shift during Remove().

Reply:
Shift down, e.g. removing element 0 means moving elements 1 through N to be 0 through N-1.